phoenix-core/src/main/java/org/apache/phoenix/coprocessor/DropColumnMutator.java

```diff
@@ -268,7 +272,19 @@ public MetaDataMutationResult validateAndAddMetadata(PTable table,
             }
 
         }
-        tableMetaData.addAll(additionalTableMetaData);
+        if (isDroppingColumns) {
+            //We're changing the application-facing schema by dropping a column, so update the DDL
+            // timestamp to current _server_ timestamp
+            if (MetaDataUtil.isTableQueryable(table.getType())) {
+                long serverTimestamp = EnvironmentEdgeManager.currentTimeMillis();
+                additionalTableMetaData.add(MetaDataUtil.getLastDDLTimestampUpdate(tableHeaderRowKey,
+                    clientTimeStamp, serverTimestamp));
+            }
+            //we don't need to update the DDL timestamp for any child views we may have, because
+            // when we look up a PTable for any of those child views, we'll take the max timestamp
+            // of the view and all its ancestors
+            tableMetaData.addAll(additionalTableMetaData);
```

Comment:
`tableMetaData.addAll(additionalTableMetaData);`
The above line won't be executed if isDroppingColumns is false. Earlier it was getting executed even if we are dropping columns.